src/query_plan/operation.rs

```diff
@@ -869,6 +891,34 @@ pub(crate) mod normalized_field_selection {
             &mut self.data.sibling_typename
         }
 
+        pub(crate) fn type_if_added_to(
+            &self,
+            parent_type: &CompositeTypeDefinitionPosition,
+            schema: &ValidFederationSchema,
+        ) -> Option<CompositeTypeDefinitionPosition> {
```

Comment:
based on the JS implementation this should be returning actual schema type (i.e. `apollo-compiler` enum `Type`)

Reply:
actually since the only place calling into this method is then unwrapping the type (`baseType`) we could probably convert the GraphQL type to a `CompositeTypeDefinitionPosition`

Reply:
> actually since the only place calling into this method is then unwrapping the type (baseType) we could probably convert the GraphQL type to a CompositeTypeDefinitionPosition

This was my thought. In the JS code, they ensure that the object is a composite type. This also avoids any (currently) unnecessary `TryInto` calls. Either way, there should be a port note explaining this.